Makefile

```diff
@@ -2180,12 +2180,18 @@ libgcc_s_dir := ../../$(aarch64_gccbase)/lib64
 endif
 
 $(out)/bootfs.bin: scripts/mkbootfs.py $(bootfs_manifest) $(bootfs_manifest_dep) $(tools:%=$(out)/%) \
-		$(out)/libenviron.so $(out)/libvdso.so $(out)/libsolaris.so
+		$(out)/libenviron.so $(out)/libsolaris.so
 	$(call quiet, olddir=`pwd`; cd $(out); "$$olddir"/scripts/mkbootfs.py -o bootfs.bin -d bootfs.bin.d -m "$$olddir"/$(bootfs_manifest), MKBOOTFS $@)
 
 $(out)/bootfs.o: $(out)/bootfs.bin
 $(out)/bootfs.o: ASFLAGS += -I$(out)
 
+$(out)/libvdso-stripped.so: $(out)/libvdso.so
+	$(call quiet, $(STRIP) $(out)/libvdso.so -o $(out)/libvdso-stripped.so, STRIP libvdso.so -> libvdso-stripped.so)
+
+$(out)/libvdso.o: $(out)/libvdso-stripped.so
+$(out)/libvdso.o: ASFLAGS += -I$(out)
```

Comment:
Nice black magic! :-)
I would consider using a different name for libvdso.o - I think it can really confuse some future reader libvdso.o isn't build from libvdso.cc (as one would assume) but rather goes libvdso.cc -> libvdso.so -> libvdso-stripped.so -> libvdso.S -> libvdso.o
I have an idea - how about renaming libvdso.S to libvdso-content.S (or something) and calling the result libvdso-content.o instead of libvdso.o?

Reply:
Agree, I will make this change and push it ASAP.

Reply:
I have updated the code with you latest suggestions.